src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/LocalizationModelDeserializer.cs

```diff
@@ -0,0 +1,230 @@
+                var instructions = LoadManualInstructionModels(postActionParts
+                    .Where(s => s.nameParts.Skip(1).FirstOrDefault().StartsWith(ManualInstructionIndexPrefix))
+                    .Select(s => (s.nameParts.Skip(1), s.localizedString)));
+
+                results[postActionIndex] = new PostActionLocalizationModel()
+                {
+                    Description = description,
+                    Instructions = instructions,
+                };
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Generates manual instruction localization models. The given parts should begin with the index string
+        /// as shown below and shouldn't include "postActions[x]".
+        /// <list type="table">
+        /// <item>manualInstructions[0].text</item>
+        /// <item>manualInstructions[1].text</item>
+        /// </list>
+        /// </summary>
+        private static IReadOnlyList<string?> LoadManualInstructionModels(IEnumerable<(IEnumerable<string> nameParts, string localizedString)> strings)
+        {
+            var results = new List<string?>();
+
+            foreach (var instructionParts in strings.GroupBy(p => p.nameParts.FirstOrDefault()))
+            {
+                if (!GetIndexFromString(instructionParts.Key, ManualInstructionIndexPrefix, ManualInstructionIndexSuffix, out int instructionIndex) ||
+                    instructionIndex > 256)
+                {
+                    // Invalid index.
+                    continue;
+                }
+
+                while (results.Count <= instructionIndex)
+                {
+                    // We haven't processed localizations for these instructions. Just put null for now.
+                    // Localization file also may not specify translations for all the manual instructions.
+                    results.Add(null);
+                }
+
+                string? text = instructionParts.FirstOrDefault(p => p.nameParts.Skip(1).SingleOrDefault() == "text").localizedString;
+                results[instructionIndex] = text;
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Parses the index from string of format &lt;prefix&gt;index&lt;suffix&gt;.
+        /// </summary>
+        /// <param name="value">The string to be parsed.</param>
+        /// <param name="prefix">The prefix that the string should begin with. Index starts after the prefix.</param>
+        /// <param name="suffix">The suffix that the string should end with. Index ends with the suffix.</param>
+        /// <returns>True if parsing succeeded. False if the string was incorrectly formatted or the index
+        /// string cannot be converted to an integer.</returns>
+        private static bool GetIndexFromString(string value, string prefix, string suffix, out int index)
```

Comment:
should it be extension instead?

Reply:
This is useful in very few cases (probably only in this file). I don't think it would be very useful as an extension method.